libmscore/rendermidi.cpp

```diff
@@ -861,57 +878,56 @@ static void collectMeasureEventsDefault(EventMap* events, Measure* m, Staff* sta
                                           }
                                     }
 
-                              // Highest velocity is a hack to stop weird, sudden changes in velocity
-                              // for the same segment and staff. We only add dynamics for one track
-                              // and the loudest track wins(?)
-                              if (velocityStart > highestVelocity) {
```

Comment:
I got rid of this hack because it was fixed by the new system that prevents duplicate velocity events.